### ISSUE TITLE
Tipper to convert "" +x to x+ ""

### ISSUE THREAD

**Issue by yossigil** [SpartanRefactoring/Main#116]
Also, fix the one that deals with x.toString() goes to x + ""

**Comment by yossigil** [comment]
Dan, I do not believe you did a real wring from start to end, so keep on this, and I will add one for you personally.

**Comment by dngreenst** [comment]
Stav has already done this, and tests to this effect pass.

**Comment by yossigil** [comment]
I did not see this in action.

**Comment by yossigil** [comment]
Pending on #119

**Comment by dngreenst** [comment]
Should work now.

**Comment by yossigil** [comment]
Then, close it.

**Comment by NivShalmon** [comment]
This seems to be bugged:
```
"" + 0 + (x - 7)
```
this turns into 0+"". I belive this is caused because the wring ignores extended operands and drops them when changing the order.

Another example @yossigil found:
Ok, here is the simplifications I get:

``` java
  private static String description(final Expression x) {
    return x == null ? "Use isEmpty()" : "Use " + x + ".isEmpty()";
  }
```
to
```java
  private static String description(final Expression x) {
    return "Use " + (x == null ? "isEmpty()" : "" + x + ".isEmpty()");
  }
```
to 
```java
  private static String description(final Expression x) {
    return "Use " + (x == null ? "isEmpty()" : x + "");
  }
```
to
```java
  private static String description(final Expression x) {
    return "Use " + (x != null ? x + "" : "isEmpty()");
  }
```
to 
```java
  private static String description(final Expression x) {
    return "Use " + (x != null ? x + "" : "isEmpty()");
  }
```
which is a bug.

@NivShalmon : can you convert this into a test case, `trimming ... to .. to.. `
sot that we can pinpoint the problem? Put this test in `@Ignore`

**Comment by yossigil** [comment]
@NivShalmon :+1:

**Comment by ravivos** [comment]
There is a bug in a unit test, please fix

**Comment by NivShalmon** [comment]
Problem solved, this test tests like 3 different tippers (since that's how we did things back then and it was a code example from a bug), so I removed the parts irrelevant to this tipper. The code example is still in this issue one of my previous comments if we ever want to rewrite the test for some reason
